### PR TITLE
Feature/devmarkt decline

### PR DESCRIPTION
**Pull request by T1Il**
It's obviously shitcode. I'll need some help w/ the code cleanup.
aka kobolds Zeit stehlen

### CODE REVIEW

File: src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt [hunk 1]
```diff
@@ -91,4 +104,100 @@ class DevmarktRequestUpdater(
 
         event.jda.httpClient.newCall(request).execute()
     }
+
+    /**
+     * Reacts if a message is received
+     */
+    @SubscribeEvent
+    fun onMessageReceived2(event: GuildMessageReceivedEvent) {
```
Comment (jkoenig134) [inline src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt]: onReceiveDenyMessage

File: src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt [hunk 2]
```diff
@@ -91,4 +104,100 @@ class DevmarktRequestUpdater(
...
+
+        val check = event.guild.getEmoteById(emoteCheckId) ?: return
+        val block = event.guild.getEmoteById(emoteBlockId) ?: return
+        val message = event.message
+
+        if (!message.contentRaw.startsWith("deny:")) {
+            return
+        }
+
+        val referencedMessage = event.message.referencedMessage ?: return
```
Comment (jkoenig134) [inline src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt]: check this before checking the message content

File: src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt [hunk 3]
```diff
@@ -91,4 +104,100 @@ class DevmarktRequestUpdater(
...
+
+        if (isNewEntryMessage(referencedMessage)) {
```
Comment (jkoenig134) [inline src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt]: bounce... if ! return

File: src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt [hunk 4]
```diff
@@ -91,4 +104,100 @@ class DevmarktRequestUpdater(
...
+
+            message.delete().queue()
+            event.jda.httpClient.newCall(request).execute()
+
+        } else if (reactionEmoteId == emoteBlockId) {
+            message.delete().queue()
+        }
+
+    }
 }
```
Comment (jkoenig134) [inline src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt]: newline

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt (hunk 1 above)
Comment (T1Il) [reply src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt]: To my defence, I thought, I'd changed that already.

File: src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt [hunk 5]
```diff
@@ -39,8 +43,20 @@ class DevmarktRequestUpdater(
         return (embeds.isNotEmpty() && "Neue Devmarkt-Anfrage" == embeds[0].title)
     }
 
+    private fun isNewReasonMessage(message: Message): Boolean {
+        val embeds = message.embeds
+        return (embeds.isNotEmpty() && "Begründung" == embeds[0].title)
+    }
+
+    private fun getFieldValue(message: Message, name: String): @Nullable String? {
```
Comment (jkoenig134) [inline src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt]: Annotation necessary?

File: src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt [hunk 6]
```diff
@@ -16,12 +16,16 @@
...
 package com.github.devcordde.devcordbot.listeners
 
+import net.dv8tion.jda.api.EmbedBuilder
 import net.dv8tion.jda.api.entities.Message
 import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
 import net.dv8tion.jda.api.events.message.react.MessageReactionAddEvent
 import net.dv8tion.jda.api.hooks.SubscribeEvent
 import okhttp3.FormBody
 import okhttp3.Request
+import java.awt.Color
```
Comment (jkoenig134) [inline src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt]: ??

File: src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt [hunk 7]
```diff
@@ -91,4 +104,99 @@ class DevmarktRequestUpdater(
...
-}
\ No newline at end of file
+
+    /**
+     * Reacts if a message is received
+     */
+    @SubscribeEvent
+    fun onReceiveDenyMessage(event: GuildMessageReceivedEvent) {
+        if (event.channel.id != modChannel
+        ) {
```
Comment (jkoenig134) [inline src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt]: closed brace in upper line, btw let IntelliJ format the file

File: src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt [hunk 8]
```diff
@@ -91,4 +104,99 @@ class DevmarktRequestUpdater(
...
+            return
+        }
+
+        val check = event.guild.getEmoteById(emoteCheckId) ?: return
+        val block = event.guild.getEmoteById(emoteBlockId) ?: return
+        val message = event.message
+        val referencedMessage = event.message.referencedMessage ?: return
+
+        if(!isNewEntryMessage(referencedMessage)) {
```
Comment (jkoenig134) [inline src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt]: could be 
```
if(!isNewEntryMessage(referencedMessage) || !message.contentRaw.startsWith("deny:")) { 
    return
}
```

File: src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt [hunk 9]
```diff
@@ -91,4 +104,99 @@ class DevmarktRequestUpdater(
...
+        val message = event.retrieveMessage().complete() ?: return
+
+        if (!isNewReasonMessage(message)) {
+            return
+        }
+
+        val reactionEmoteId = event.reactionEmote.id
+        val requestId = getFieldValue(message, "Request-ID") ?: return
+
+        if (reactionEmoteId == emoteCheckId) {
```
Comment (jkoenig134) [inline src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt]:
```
if (reactionEmoteId == emoteBlockId) {
    return message.delete().queue()
}

if (reactionEmoteId != emoteCheckId) {
  return
}

...
```

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt (hunk 5 above)
Comment (T1Il) [reply src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt]: YES

Comment (jkoenig134) [reply src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt]: The ? makes the String nullable..

Comment (T1Il) [reply src/main/kotlin/com/github/devcordde/devcordbot/listeners/DevmarktRequestUpdater.kt]: IntelliJ says no

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**